### PR TITLE
Feat #593 DomainOfExpertise / owner - drop down list (combobox) improvements

### LINKED ISSUE #593: DomainOfExpertise / owner - drop down list  (combobox) improvements
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
as an example in the commonfilestore - add a Folder

  - [x] order the domains of expertise in alphabetical order so its easier to find what you are looking for
  - [x] show both the name and shortname: name [shortname]
  - [x] when creating something new, preselect the domain that is active in the current session
  - [x] format the current active domain with bold

### Logs

  - [ ] There are no logs regarding the issue
  - [ ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by joao4all**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Closes #593 DomainOfExpertise / owner - drop down list (combobox) improvements

### CODE REVIEW

File: COMET.Web.Common.Tests/ViewModels/Components/Selectors/DomainOfExpertiseSelectorViewModelTestFixture.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            sessionService.Setup(x => x.GetSiteDirectory()).Returns(siteDirectory);
+            sessionService.Setup(x => x.GetDomainOfExpertise(It.IsAny<Iteration>())).Returns(new DomainOfExpertise());
+
+            this.viewModel = new DomainOfExpertiseSelectorViewModel(sessionService.Object, this.messageBus);
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            this.viewModel.Dispose();
```
Comment (antoineatstariongroup) [inline COMET.Web.Common.Tests/ViewModels/Components/Selectors/DomainOfExpertiseSelectorViewModelTestFixture.cs:77]: please dispose the messageBus also

File: COMET.Web.Common/Components/Selectors/DomainOfExpertiseSelector.razor.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    using COMET.Web.Common.ViewModels.Components.Selectors;
+
+    using Microsoft.AspNetCore.Components;
+
+    /// <summary>
+    /// Component used to select a <see cref="DomainOfExpertise" />
+    /// </summary>
+    public partial class DomainOfExpertiseSelector
+    {
+		/// <summary>
```
Comment (antoineatstariongroup) [inline COMET.Web.Common/Components/Selectors/DomainOfExpertiseSelector.razor.cs:38]: Identation issue in the file

File: COMET.Web.Common/Components/Selectors/DomainOfExpertiseSelector.razor.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		[Parameter]
+        public string DisplayText { get; set; } = "Select a Domain:";
+
+        /// <summary>
+        /// The css class used to apply custom styles to the selector
+        /// </summary>
+        [Parameter]
+        public string CssClass { get; set; }
+
+        private static string GetDomainNameAndShortnameToDisplay(DomainOfExpertise domainOfExpertise)
```
Comment (antoineatstariongroup) [inline COMET.Web.Common/Components/Selectors/DomainOfExpertiseSelector.razor.cs:56]: Missing doc

File: COMETwebapp/ViewModels/Components/SiteDirectory/UserManagement/UserManagementTableViewModel.cs [hunk 4]
```diff
@@ -30,12 +30,14 @@ namespace COMETwebapp.ViewModels.Components.SiteDirectory.UserManagement
...
 
     using COMET.Web.Common.Services.SessionManagement;
+    using COMET.Web.Common.ViewModels.Components.Selectors;
 
     using COMETwebapp.Services.ShowHideDeprecatedThingsService;
     using COMETwebapp.ViewModels.Components.Common.DeprecatableDataItemTable;
     using COMETwebapp.ViewModels.Components.SiteDirectory.Rows;
 
     using DevExpress.Blazor;
+    using Microsoft.AspNetCore.Components;
```
Comment (antoineatstariongroup) [inline COMETwebapp/ViewModels/Components/SiteDirectory/UserManagement/UserManagementTableViewModel.cs:40]: Missing blank line

**Review by antoineatstariongroup — CHANGES_REQUESTED** [review]

File: COMET.Web.Common/Components/Selectors/DomainOfExpertiseSelector.razor.cs (hunk 2 above)
Comment (joao4all) [reply COMET.Web.Common/Components/Selectors/DomainOfExpertiseSelector.razor.cs:38]: image
Just ran resharper code cleanup to make sure it's ok, but everythings seems fine in VS

**Review by antoineatstariongroup — APPROVED** [review]

**Pull request merged**